master/internal/kubernetes/spec.go

```diff
@@ -346,16 +346,16 @@ func (p *pod) createPodSpecForGC(ctx *actor.Context) error {
 	return nil
 }
 
-func configureUniqueName(t tasks.TaskSpec, rank int) string {
+func configureUniqueName(t tasks.TaskSpec) string {
 	uniqueName := petName.Generate(2, "-")
 	switch {
 	case t.StartCommand != nil:
 		return fmt.Sprintf("cmd-%s-%s", t.TaskID, uniqueName)
 	case t.StartContainer != nil:
 		return fmt.Sprintf(
-			"exp-%d-trial-%d-%d-%s",
+			"exp-%d-trial-%d-%s",
 			t.StartContainer.InitialWorkload.ExperimentID,
-			t.StartContainer.InitialWorkload.TrialID, rank,
+			t.StartContainer.InitialWorkload.TrialID,
```

Comment:
I can print the rank in `trial.go` since I removed the rank concept from the resource providers. `rank` is more of a concept that is being used outside the scheduler. The scheduler doesn't and shouldn't care which assignment is the leader.

Reply:
> The scheduler is actually the one which implicitly assigns rank to assignments when it creates them

I changed that to be the order of containers connected. why is it necessary to let the scheduler implicitly handles that?

Reply:
changed that to be the previous state